### PR TITLE
Remove use BasicStructure from local acceptance test context

### PR DESCRIPTION
**Pull request by phil-davis**
and use methods from core ``FeatureContext`` instead.

### CODE REVIEW

**Comment by phil-davis** [comment]
Note: this will fail today, because ``getDavPath`` is private in ``FeatureContext``. That is fixed in a core PR and will appear in the core QA tarball overnight.

File: tests/acceptance/features/bootstrap/CustomGroupsContext.php [hunk 1]
```diff
@@ -135,103 +179,150 @@ public function customGroupShouldNotExist($customGroup){
...
 		$client->setThrowExceptions(true);
 		$appPath = '/customgroups/groups/';
-		$fullUrl = $this->getBaseUrl() . '/' . $this->davPath . $appPath . $customGroup;
+		$fullUrl = $this->featureContext->getBaseUrl() . '/' . $this->featureContext->getDavPath() . $appPath . $customGroup;
 		try {
-			$response = $client->proppatch($fullUrl, $properties, 1);
-			$this->response = $response;
-			return $response;
-		} catch (\Sabre\HTTP\ClientException $e) {
-			$this->response = null;
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/CustomGroupsContext.php]: how do we set the response in `BasicStructure` now? don't we need `$this->featureContext->setResponse($response);`?

File: tests/acceptance/features/bootstrap/CustomGroupsContext.php [hunk 2]
```diff
@@ -135,103 +179,150 @@ public function customGroupShouldNotExist($customGroup){
...
+	 * @return bool
+	 * @throws ClientException
+	 */
 	public function sendProppatchToCustomGroupMember($userRequesting, $customGroup, $userRequested, $properties = null){
-		$client = $this->getSabreClient($userRequesting);
+		$client = $this->featureContext->getSabreClient($userRequesting);
 		$client->setThrowExceptions(true);
 		$appPath = '/customgroups/groups/';
-		$fullUrl = $this->getBaseUrl() . '/' . $this->davPath . $appPath . $customGroup . '/' . $userRequested;
+		$fullUrl = $this->featureContext->getBaseUrl() . '/' . $this->featureContext->getDavPath() . $appPath . $customGroup . '/' . $userRequested;
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/CustomGroupsContext.php]: please make that line shorter

File: tests/acceptance/features/bootstrap/CustomGroupsContext.php [hunk 3]
```diff
@@ -135,103 +179,150 @@ public function customGroupShouldNotExist($customGroup){
...
 		try {
-			$response = $client->proppatch($fullUrl, $properties, 1);
-			$this->response = $response;
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/CustomGroupsContext.php]: same as above, don't we need to change `BasicStructure` `$response`?

File: tests/acceptance/features/bootstrap/CustomGroupsContext.php (hunk 1 above)
Comment (phil-davis) [reply tests/acceptance/features/bootstrap/CustomGroupsContext.php]: This is strange code which does a Sabre client thing. The responses from that are not the same thing as in FeatureContext. The previous code would sometimes make a mess of FeatureContext response, but it was just lucky that steps afterward did not try to use the FeatureContext method to check the HTTP response.

File: tests/acceptance/features/bootstrap/CustomGroupsContext.php (hunk 3 above)
Comment (phil-davis) [reply tests/acceptance/features/bootstrap/CustomGroupsContext.php]: See comment above - the return value of proppatch is not the sort of response that FeatureContext has.

**Comment by phil-davis** [comment]
@individual-it I pushed another commit that makes the code match the acceptance test code style "standard".

**Pull request merged**